webpack.config.js

```diff
@@ -0,0 +1,152 @@
+var path = require('path');
+
+module.exports = (env, argv) => {
+  let devtool = false;
+  if (argv.mode === 'development') {
+    devtool = 'inline-source-map';
+  }
+  console.log(`${argv.mode} build`);
+  const externals = {
+    aframe: {
+      commonjs: 'aframe',
+      commonjs2: 'aframe',
+      amd: 'aframe',
+      root: 'AFRAME' // indicates global variable
+    },
+    three: {
+      commonjs: 'three',
+      commonjs2: 'three',
+      amd: 'three',
+      root: 'THREE' // indicates global variable
+    }
+  };
+  const module = {
+    rules: [
+      {
+        test: /\.worker\.js$/,
+        use: { loader: 'worker-loader' }
+      }
+    ]
+  };
+
+  return [{
+    name: 'default',
+    devtool,
+    entry: './aframe/src/index.js',
+    output: {
+      library: 'ARjs',
+      path: path.resolve(__dirname, 'aframe/build'),
+      filename: 'aframe-ar.js',
+      libraryTarget: 'umd',
+      globalObject: 'this'
+    },
+    resolve: {
+      alias: {
+        jsartoolkit: 'artoolkit5-js',
+        threexArmarkercontrols$: path.resolve(__dirname, 'three.js/src/threex/arjs-markercontrols.js')
```

Comment:
For what reason this alias? What advantages and consequence imply?

Reply:
Good question :D . On the master branch, builds always came in pairs: "normal" an "nft". Each of them used slightly different sources for `ArMarkerControls`. I the "cat build-system", the contents of `ArMarkerControls` would have been determined by an assignment to it in whatever file had been `cat`'ed before you'd use the actual variable again. Such way of doing things offered a flexibility that you don't have out of the box in an import-based system. Webpack's `alias`-functionality offered a nice way out here to assign something different to `ArMarkerControls` based on if it's building an nft- or non-nft build.
As mentioned in https://github.com/AR-js-org/AR.js/pull/116#issuecomment-803301196 ("nft vs non-nft"), I'm still not entirely sure how the nft- vs non-nft split came in the codebase and if we still actually need it. In any case, I tried to keep things as they used to be. Please clarify further here.

Reply:
nft and non-nft come out because we used two different builds for jsartoolkit5 because we cannot solve a display issue (3d object shifted and distorted) with pattern marker using the new jsartoolkit5 lib with nft. So we used the old jsartoolkit5 lib for pattern-marker and the new one for nft. You can see that in the old cat MAkefile https://github.com/AR-js-org/AR.js/blob/f5efd21215a796a2e32b919259e4f5ce839e4b29/three.js/Makefile#L6-L7 artoolkit.min.js and artoolkit.api.js both are bundled. that is necessary to avoid that issue. Instead for nft we use the new one with nft: https://github.com/AR-js-org/AR.js/blob/f5efd21215a796a2e32b919259e4f5ce839e4b29/three.js/Makefile#L24